software-containers/README.md

```diff
@@ -16,3 +16,156 @@ group:      sss files
 
 ### xterm
 `xterm` must be installed on the container
+## Creating a CASA container
+This section describes the process of creating a CASA container. The currently supported CASA containers are in the arcade-casa folder. The process iterates over build, deploy and test.
+### Building A Container
+To build a container, go into the relevant folder with the build files or create a new folder if necessary. There are two main files to building a container: Makefile and Dockerfile.
+#### Makefile
+This is the main file that controls the build process. This file is updated to add/change/remove a CASA version or to change the build process. It invokes docker to build the container.
+#### Dockerfile
+This file controls the building of the CASA container. For example, for versions 4.3.x to 4.4.x, the ADMIT package is not added to the container. Changes to the container are made in this file.
+#### How To Build
+1. cd into the relevant folder with the build file or create a new folder if necessary
+2. update or create the relevant files, e.g. Makefile, Dockerfile, download.sh
+3. execute 'make'
+4. execute 'docker image ls'
+
+Step (3) generates a docker image in the local machine, which is listed in step(4). Here is an example list of generated docker images:
+| REPOSITORY | TAG | IMAGE ID | CREATED | SIZE |
+| --------------------- | ------- | ---------------- | --------------- | ------- |
+| bucket.canfar.net/arcade-casa | 4.5.3-el6 | 0b8c5890921b | 13 hours ago | 4.72GB |
+| bucket.canfar.net/arcade-casa | 4.5.2-el6 | 21719143194a | 13 hours ago | 4.72GB |
+
+For work that does not require a UI, for example adding a new package to CASA, we may be able to run the container and test our work at this point. Execute 'docker run -it <docker image ID> /bin/bash' and check if the work is correct.
+### Deploying A Container
+Since a CASA container runs in Arcade under Kubernetes, the container image generated from the build step needs to be pushed to the repository that Kubernetes uses.
```

Comment:
The steps in "Deploying a container" are for CADC only so should not be described here.  They're good for your own notes but not the right information for open-source collaborators.

Reply:
Done